napi.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+var log = require('npmlog')
+var path = require('path')
+var pkg = require(path.resolve('package.json'))
+
+var versionArray = process.version
+  .substr(1)
+  .replace(/-.*$/, '')
+  .split('.')
+  .map(function (item) {
+    return +item
+  })
+
+module.exports.isNapiRuntime = function (runtime) {
+  return runtime === 'napi'
+}
+
+module.exports.isSupportedVersion = function (napiVerison) {
+  var version = parseInt(napiVerison, 10)
+  return version <= getNapiVersion() && packageSupportsVersion(version)
+}
+
+var packageSupportsVersion = function (napiVerison) {
+  if (pkg.binary && pkg.binary.napi_versions) {
```

Comment:
Is this `binary` property described / standardized anywhere?

Reply:
It conforms with the way it's used for [`node-pre-gyp`](https://github.com/mapbox/node-pre-gyp#the-napi_versions-array-property). It my knowledge, these are the only two places it is currently used. I did the PR to add N-API support to `node-pre-gyp`.